CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+endif ()
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  add_compile_definitions(__LINUX__)
+elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  add_compile_definitions (__MAC_OSX__)
+  set (ENV{PKG_CONFIG_PATH} "/usr/local/lib/pkgconfig")
+elseif (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  add_compile_definitions(__WIN_32__ _WIN32_WINNT=0x0600 WIN32_LEAN_AND_MEAN)
+  include_directories("C:/Program Files/JACK2/include")
+  set (CMAKE_PREFIX_PATH "C:/Qt/5.15.2/mingw81_64/lib/cmake")
+endif ()
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Linux" OR ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(JACK REQUIRED IMPORTED_TARGET jack)
+endif ()
+
+# Find includes in corresponding build directories
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+# Instruct CMake to run moc automatically when needed.
+set(CMAKE_AUTOMOC ON)
+# Instruct CMake to create code from Qt designer ui files
+set(CMAKE_AUTOUIC ON)
+
+set(CMAKE_AUTORCC ON)
+
+# Find the QtWidgets library
+if (NOT nogui)
+  find_package(Qt5Widgets CONFIG REQUIRED)
+endif ()
+find_package(Qt5Network CONFIG REQUIRED)
+
+set(qjacktrip_SRC
+  src/main.cpp
+  src/Settings.cpp
+  src/jacktrip_globals.cpp
+  src/JackTrip.cpp
+  src/UdpHubListener.cpp
+  src/JackTripWorker.cpp
+  src/DataProtocol.cpp
+  src/UdpDataProtocol.cpp
+  src/AudioInterface.cpp
+  src/JackAudioInterface.cpp
+  src/JMess.cpp
+  src/LoopBack.cpp
+  src/PacketHeader.cpp
+  src/RingBuffer.cpp
+  src/JitterBuffer.cpp
+  src/Compressor.cpp
+  src/Limiter.cpp
+  src/Reverb.cpp
+  src/AudioTester.cpp
+  src/Patcher.cpp
+  src/SslServer.cpp
+  src/Auth.cpp
+)
+
+if (NOT nogui)
+  set (qjacktrip_SRC ${qjacktrip_SRC}
+    src/gui/qjacktrip.cpp
+    src/gui/about.cpp
+    src/gui/messageDialog.cpp
+    src/gui/qjacktrip.qrc
+  )
+
+  if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+    set (qjacktrip_SRC ${qjacktrip_SRC} win/qjacktrip.rc)
+  elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set (qjacktrip_SRC ${qjacktrip_SRC} src/gui/NoNap.mm)
+    set (CMAKE_C_FLAGS "-x objective-c")
+    set (CMAKE_EXE_LINKER_FLAGS "-framework Foundation")
+  endif ()
+endif ()
+
+add_compile_definitions(WAIRTOHUB)
+
+# Tell CMake to create the executable
+if (nogui)
+  add_executable(jacktrip ${qjacktrip_SRC})
+else ()
+  add_executable(qjacktrip ${qjacktrip_SRC})
+endif ()
+
+# Use the Widgets module from Qt 5.
+if (${CMAKE_SYSTEM_NAME} MATCHES "Linux" OR ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  if (nogui)
+    target_link_libraries(jacktrip Qt5::Network PkgConfig::JACK)
+  else ()
+    target_link_libraries(qjacktrip Qt5::Widgets Qt5::Network PkgConfig::JACK)
+    if (NOT EXISTS "${CMAKE_BINARY_DIR}/jacktrip")
+      add_custom_target(create_symlink ALL COMMAND ${CMAKE_COMMAND} -E create_symlink qjacktrip jacktrip)
+    endif ()
+  endif ()
+elseif (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  if (nogui)
+    target_link_libraries(jacktrip Qt5::Network "C:/Program Files/JACK2/lib/libjack64.lib" ws2_32)
```

Comment:
I should look into how to do that. I've always just prioritised getting cmake working on my own system (so I can integrate things with KDevelop easily) and having the build script work more universally, but it would be nice for other users to have options.

Reply:
And done

Reply:
Thanks!